### PR TITLE
add an example for qwen doc QA with langchain + llamaindex

### PR DESCRIPTION
**Pull request by yhxx511**
add an ipynb example for qwen doc search QA scenario  by using langchain with llamaindex RAG implementation

### CODE REVIEW

**Review by wenmengzhou — COMMENTED** [review]
it is better to clear the output of notebook cells

File: examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb [hunk 1]
```diff
@@ -2,278 +2,296 @@
...
    "source": [
-    "def search_relevant_context(question, topk=1, client=dashvector_client):\n",
-    "    # query and recall the relevant information\n",
-    "    collection = client.get(collection_name)\n",
-    "\n",
-    "    # recall the top k similarity results from DashVector\n",
-    "    rsp = collection.query(generate_embeddings(question), output_fields=['raw'],\n",
-    "                           topk=topk)\n",
-    "    return \"\".join([item.fields['raw'] for item in rsp.output])"
+    "!pip install pypdf langchain unstructured transformers_stream_generator\n",
```
Comment (wenmengzhou) [inline examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb:277]: specify the version of theses dependencies to make it robust to run in anytime

File: examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb [hunk 2]
```diff
@@ -282,153 +300,386 @@
...
-    "tags": []
-   },
-   "outputs": [],
-   "source": [
-    "# initialize qwen 7B model\n",
-    "from modelscope import AutoModelForCausalLM, AutoTokenizer\n",
-    "from modelscope import GenerationConfig\n",
+    "!mkdir -p /root/nltk_data/tokenizers\n",
+    "!mkdir -p /root/nltk_data/taggers\n",
+    "!cp /mnt/workspace/punkt.zip /root/nltk_data/tokenizers\n",
```
Comment (wenmengzhou) [inline examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb:356]: it is better to provide the url of these files and execute `wget` to download to make sure this cell can run successfully

File: examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb [hunk 3]
```diff
@@ -282,153 +300,386 @@
...
     }
    ],
    "source": [
-    "# test the case with knowledge\n",
-    "context = search_relevant_context(question, topk=1)\n",
-    "answer = answer_question(question, context)\n",
-    "print(f'question: {question}\\n' f'answer: {answer}')"
+    "import os\n",
+    "from abc import ABC\n",
+    "from typing import Any, List, Optional, Dict, cast\n",
```
Comment (wenmengzhou) [inline examples/pytorch/application/qwen_doc_search_QA_based_on_dashscope.ipynb:475]: this cell is too long to read, pls split this cell to multiple cells at each location when you need to write comment, and create a markdown block to write that comment

**Comment by yhxx511** [comment]
ignore the changed for dashscope.ipynb, it's been unexpectedly changed and got reverted by force-push.

**Pull request merged**